go/fn/object_test.go

```diff
@@ -10,6 +10,86 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+func TestGVK(t *testing.T) {
```

Comment:
Suggest adding the cases where a resource cannot match.

Reply:
The cannot match situations are tested here as well. I tested several matching possibilities with the same input.

Reply:
I mean the current test cases cover
1. GVK A matches GVK X
2. GVK A does not match GVK X
can we have test cases where
3. GVK A and B, only A matches GVK X while B does not? The difference is that the same matching rule can distinguish different resources.